python/taichi/lang/ast/ast_transformer.py

```diff
@@ -762,16 +762,30 @@ def build_Module(ctx, node):
 
     @staticmethod
     def build_Attribute(ctx, node):
-        if node.attr == "append" and isinstance(node.value, ast.Subscript):
+        is_subscript = isinstance(node.value, ast.Subscript)
+        if is_subscript:
             x = build_stmt(ctx, node.value.value)
-            if not isinstance(x, Field) or x.parent(
-            ).ptr.type != _ti_core.SNodeType.dynamic:
-                raise TaichiSyntaxError(
-                    f"In Taichi scope the `append` method is only defined for dynamic SNodes, but {x} is encountered"
-                )
-            index = build_stmt(ctx, node.value.slice)
-            node.value.ptr = None
-            node.ptr = lambda val: append(x.parent(), index, val)
+        else:
+            x = build_stmt(ctx, node.value)
+
+        names = ("append", "deactivate")
+        if isinstance(x, Field) and x.parent(
+        ).ptr.type == _ti_core.SNodeType.dynamic and node.attr in names:
+
+            def build_dynamic_snode_hack():
+                if is_subscript:
+                    index = build_stmt(ctx, node.value.slice)
+                else:
+                    index = []
+
+                node.value.ptr = None
+                if node.attr == "append":
+                    node.ptr = lambda val: append(x.parent(), index, val)
+                else:
+                    node.ptr = lambda: deactivate(x.parent(), index)
+
+            build_dynamic_snode_hack()
+
         else:
             build_stmt(ctx, node.value)
```

Comment:
Emmm currently we can't call `build_stmt` on a node twice... I'll file a PR that enables us to call `build_stmt` on a node more than once immediately. Please wait for the PR.

Reply:
emmm I found that I can't do that because we have static for...

Reply:
We have to make sure that a node is built at most once for now...